### PR TITLE
Don't pass --with-build-sysroot with NATIVE build

### PR DESCRIPTION
**Pull request by michaelforney**
This is not needed since we already have a native musl toolchain at our
disposal. Previously, since this flag was passed for NATIVE=1, the build would
fail because the build sysroot is only built for cross toolchains.

### CODE REVIEW

**Comment by richfelker** [comment]
Can you explain how/where it was failing without this? Building with NATIVE set works for me, so maybe the way it's breaking is more subtle?

**Comment by michaelforney** [comment]
The error occurs when building the target libgcc using the system compiler (with a bogus `--sysroot`).

Here is a transcript that hopefully that explains the issue enough: http://sprunge.us/WjfB

As the transcript indicates, removing `--with-build-sysroot` or building the sysroot targets manually both result in a successful build.

Maybe your system's x86_64-linux-musl-gcc searches for limits.h in more places than mine? Mine's a pretty standard compiler built with musl-cross-make.

This is also quite related to another issue I came across with musl-cross-make. If I build with `NATIVE=1 COMMON_CONFIG=--build=x86_64-linux-musl` so that $build == $host == $target, it fails when trying to build the target libgcc with xgcc because xgcc is built with a relocatable sysroot, so when it is invoked from the build directory, it detects that it has been relocated and searches for system headers relative to itself (and fails). I think this a fundamental problem with building gcc with a relocatable sysroot and without --build-sysroot. I worked around it by passing `--build-sysroot=/` in my COMMON_CONFIG, but I don't think we can do this in general without making assumptions about the user's musl compiler.

So, to summarize, we shouldn't set the build sysroot to something that doesn't exist for NATIVE=1. We could either not set it (so the system toolchain's headers and libraries are used) or build the sysroot regardless of NATIVE.

**Comment by michaelforney** [comment]
I rebased on top of the current master, and confirmed that the issue still exists with gcc 6.2.0.

I also think I figured out why you aren't seeing this same failure. If I build a cross toolchain (`NATIVE` unset), and then use that to build a native toolchain (`NATIVE=1`), I do not see any failure. This is because gcc is able to find the headers at `/path/to/toolchain/bin/../lib/gcc/x86_64-linux-musl/6.2.0/../../../../x86_64-linux-musl/include/limits.h` (`/path/to/toolchain/x86_64-linux-musl/include/limits.h`).

However, if I use that native compiler, and attempt to build another native compiler, I get the error reported above because only `/path/to/toolchain/bin/../lib/gcc/x86_64-linux-musl/6.2.0/include/limits.h` is searched (as well as the empty sysroot).

See these two strace logs for *identical* toolchain builds, except for `NATIVE=1`:

Cross: http://sprunge.us/DjiX
Native: http://sprunge.us/OMIf

In any case, I don't think either of these two proposed patches should be controversial. The current code does not build a sysroot if `NATIVE=1`. If there is no sysroot, then we should not pass `--with-build-sysroot` to configure.

**Comment by richfelker** [comment]
I'm not sure if this will be correct with future changes I want to make, undoing the combined-tree gcc+binutils and less-hackish invocation of the gcc makefile to do the single-stage build. However, I agree with you that the changes are reasonable for the current way we're doing things, and I'll keep in mind the problems you were experiencing when I get around to making the big overhauls I want to do. Merging. Thanks for explaining the motivations.
